downstream.go

```diff
@@ -777,6 +777,15 @@ func (dc *downstreamConn) welcome() error {
 		Params:  []string{dc.nick, "No MOTD"},
 	})
 
+	if uc := dc.upstream(); uc != nil && uc.nick != dc.nick {
```

Comment:
Maybe we should have an `updateNick` function that does this, similar to `updateSupportedCaps`?